python/cudf/cudf/core/column/numerical.py

```diff
@@ -320,13 +322,45 @@ def find_and_replace(self, to_replace, replacement, all_nan):
         replacement_col = column.as_column(
             replacement, dtype=replacement_dtype
         )
+
+        # If to_replace is a python list or scalar, check mininum viable dtype
+        if isinstance(to_replace, list) or is_scalar(to_replace):
+            to_replace_dtype = min_numeric_column_type(to_replace_col)
+        else:
+            to_replace_dtype = to_replace.dtype
+        if to_replace_dtype > self.dtype or (
+            np.issubdtype(self.dtype, np.number)
+            and np.issubdtype(to_replace_dtype, np.floating)
```

Comment:
wanted to get rid of the `or` case but didn't find a way around it because of these cases

| dtype1  | operator | dtype2 | output |
|---------|----------|--------|--------|
| float32 | >        | int32  | False  |
| float32 | <        | int32  | False  |
| float32 | <        | int64  | False  |
| float32 | >        | int64  | False  |
| float64 | >        | int64  | True   |